components/scream/src/share/io/tests/io.cpp

```diff
@@ -251,116 +264,265 @@ TEST_CASE("input_output_basic","io")
+  ekat::parse_yaml_file("io_test_control.yaml",control_params);
+  EKAT_REQUIRE_MSG(control_params.isSublist(output_freq_units),"ERROR! output frequency units " + output_freq_units + " does not have an entry in io_test_control.yaml");
+  const auto& freq_params = control_params.sublist(output_freq_units);
+  const Int max_steps = freq_params.get<int>("Max Steps");
+  const Int dt        = freq_params.get<int>("dt");
+
+  {
+    util::TimeStamp time = t0;
+
+    // Re-create the fm anew, so the fields are re-inited for each output type
+    auto field_manager = get_test_fm(grid);
+    field_manager->init_fields_time_stamp(t0);
+    // Set up parameter list control for output
+    ekat::ParameterList params;
+    ekat::parse_yaml_file("io_test_template.yaml",params);
+    params.set<std::string>("Averaging Type",output_type);
+    auto& params_sub = params.sublist("Output Control");
+    params_sub.set<std::string>("Frequency Units",output_freq_units);
+    io_control.frequency = params_sub.get<int>("Frequency");
+    // Set up output manager.
+    OutputManager om;
+    om.setup(io_comm,params,field_manager,gm,t0,t0,false);
+    io_comm.barrier();
+
+    const auto& out_fields = field_manager->get_groups_info().at("output");
+    // Time loop
+    for (Int ii=0;ii<max_steps;++ii) {
+      // Update the fields
+      for (const auto& fname : out_fields->m_fields_names) {
+        auto f  = field_manager->get_field(fname);
+        f.sync_to_host();
+        auto fl = f.get_header().get_identifier().get_layout();
+        switch (fl.rank()) {
+          case 1:
+            {
+              auto v = f.get_view<Real*,Host>();
+              for (int i=0; i<fl.dim(0); ++i) {
+                v(i) += dt;
+              }
+            }
+            break;
+          case 2:
+            {
+              auto v = f.get_view<Real**,Host>();
+              for (int i=0; i<fl.dim(0); ++i) {
+                for (int j=0; j<fl.dim(1); ++j) {
+                  v(i,j) += dt;
+                }
+              }
+            }
+            break;
+          default:
+            EKAT_ERROR_MSG ("Error! Unexpected field rank.\n");
+        }
+        f.sync_to_dev();
+      }
+
+      // Run the output manager for this time step
+      time += dt;
+      om.run(time);
+      if (io_control.is_write_step(time)) {
+        output_stamps.push_back(time.to_string());
+        io_control.nsteps_since_last_write = 0;
+        io_control.timestamp_of_last_write = time;
+      }
     }
+
+    // Finalize the output manager (close files)
+    om.finalize();
   }
-  for (int jj=0;jj<num_levs;++jj) {
-    avg_val = (max_steps+1)/2.0*dt + (jj+1)/10.;  //note x0=(jj+1)/10 in this case.
-    REQUIRE(std::abs(f2_host(jj)-avg_val)<tol);
-  }
-  avg_input.finalize();
+
+  // Get a fresh new field manager
+  auto field_manager = get_test_fm(grid);
+  const auto& out_fields = field_manager->get_groups_info().at("output");
+  auto reset_fields = [&] () {
```

Comment:
This utility is no longer needed, since you run _one_ avg type at a time in the run function.